tools/c7n_traildb/trailes.py

```diff
@@ -0,0 +1,277 @@
+# Copyright 2017 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+import logging
+import os
+import subprocess
+import thread
+
+import click
+from concurrent.futures import ProcessPoolExecutor, as_completed
+from dateutil.parser import parse as parse_date
+from elasticsearch import Elasticsearch, helpers, RequestsHttpConnection
+import jsonschema
+import sqlite3
+
+import yaml
+
+from c7n.credentials import assumed_session
+from c7n.executor import ThreadPoolExecutor
+from c7n.utils import local_session
+
+MAX_POINTS = 1440.0
+NAMESPACE = 'CloudMaid'
+
+log = logging.getLogger('c7n.trailes')
+
+CONFIG_SCHEMA = {
+    'type': 'object',
+    'additionalProperties': True,
+    'properties': {
+        'indexer': {
+            'type': 'object',
+            'required': ['host', 'port', 'idx_name'],
+            'properties': {
+                'host': {'type': 'string'},
+                'port': {'type': 'number'},
+                'user': {'type': 'string'},
+                'password': {'type': 'string'},
+                'idx_name': {'type': 'string'},
+                'proxy': {'type': 'string'},
+            },
+            'additionalProperties': True
+        },
+        'accounts': {
+            'type': 'array',
+            'items': {
+                'type': 'object',
+                'required': ['name', 'bucket', 'regions', 'title'],
+                'properties': {
+                    'name': {'type': 'string'},
+                    'title': {'type': 'string'},
+                    'tags': {'type': 'object'},
+                    'bucket': {'type': 'string'},
+                    'regions': {'type': 'array', 'items': {'type': 'string'}}
+                }
+            }
+        }
+    }
+}
+
+
+class MyConnection(RequestsHttpConnection):
+    def __init__(self, *args, **kwargs):
+        proxies = kwargs.pop('proxies', {})
+        super(MyConnection, self).__init__(*args, **kwargs)
+        self.session.proxies = proxies
+
+
+def get_es_client(config):
+    host = [config['indexer'].get('host', 'localhost')]
+    es_kwargs = {}
+    user = config['indexer'].get('user', False)
+    password = config['indexer'].get('password', False)
+    if user and password:
+        es_kwargs['http_auth'] = (user, password)
+
+    es_kwargs['port'] = config['indexer'].get('port', 9200)
+
+    proxy = config['indexer'].get('proxy', False)
+    if proxy:
```

Comment:
per the es resource comments on the same, proxy inherits in most languages from standard env vars, rather than making this additional config, and using a separate connection class, just omit and you'll get the standard behavior.